modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -617,117 +605,65 @@ object SeqexecEngine extends SeqexecConfiguration {
-    val tcsControl              = cfg.require[ControlStrategy]("seqexec-engine.systemControl.tcs")
-    val odbNotifications        = cfg.require[Boolean]("seqexec-engine.odbNotifications")
-    val gpiUrl                  = tag[GpiSettings][Uri](cfg.require[Uri]("seqexec-engine.gpiUrl"))
-    val ghostUrl                = tag[GhostSettings][Uri](cfg.require[Uri]("seqexec-engine.ghostUrl"))
-    val gpiGDS                  = tag[GpiSettings][Uri](cfg.require[Uri]("seqexec-engine.gpiGDS"))
-    val ghostGDS                = tag[GhostSettings][Uri](cfg.require[Uri]("seqexec-engine.ghostGDS"))
-    val instForceError          = cfg.require[Boolean]("seqexec-engine.instForceError")
-    val failAt                  = cfg.require[Int]("seqexec-engine.failAt")
-    val odbQueuePollingInterval = cfg.require[Duration]("seqexec-engine.odbQueuePollingInterval")
-    val tops                    = decodeTops(cfg.require[String]("seqexec-engine.tops"))
-    val caAddrList              = cfg.lookup[String]("seqexec-engine.epics_ca_addr_list")
-    val ioTimeout               = cfg.require[Duration]("seqexec-engine.ioTimeout")
-
-    // TODO: Review initialization of EPICS systems
-    def initEpicsSystem(sys: EpicsSystem[_], tops: Map[String, String]): IO[Unit] =
-      IO.apply(
-        Option(CaService.getInstance()) match {
-          case None => throw new Exception("Unable to start EPICS service.")
-          case Some(s) =>
-            sys.init(s, tops).leftMap {
-                case SeqexecFailure.SeqexecException(ex) => throw ex
-                case c: SeqexecFailure                   => throw new Exception(SeqexecFailure.explain(c))
-            }
+  /**
+   * Build the seqexec and setup epics
+   */
+  def build(
+    site: Site,
+    systems: Systems[IO],
+    conf: SeqexecEngineConfiguration,
+    metrics: SeqexecMetrics)(
+    implicit cs: ContextShift[IO],
+             tio: Timer[IO],
+             L: Logger[IO]
+  ): IO[SeqexecEngine] = createTranslator(site, systems, conf).flatMap{ translator =>
+      // TODO: Review initialization of EPICS systems
+      def initEpicsSystem(sys: EpicsSystem[_], tops: Map[String, String]): IO[Unit] =
+        IO.apply(
+          Option(CaService.getInstance()) match {
+            case None => throw new Exception("Unable to start EPICS service.")
+            case Some(s) =>
+              sys.init(s, tops).leftMap {
+                  case SeqexecFailure.SeqexecException(ex) => throw ex
+                  case c: SeqexecFailure                   => throw new Exception(SeqexecFailure.explain(c))
+              }
+          }
+        ) *> IO.unit
+
+      val tops = decodeTops(conf.tops)
+
+      // Ensure there is a valid way to init CaService either from
+      // the configuration file or from the environment
+      val caInit   = conf.epicsCaAddrList.map(a => IO.apply(CaService.setAddressList(a))).getOrElse {
```

Comment:
`IO.apply(...)` → `IO(...)` here and elsewhere.

Reply:
👍 Hopefully soon it will be `Sync[F].delay` instead